### PR TITLE
sim: add sqlite to further analyze sim runs

### PR DESCRIPTION
**Pull request by czarcas7ic**
## What is the purpose of the change

This PR introduces basic sqlite functionality to the simulator. Once this gets approved, we can gather more data as well as provide make commands to show basic trends with the provided dataset. 

If `WriteStatsToDB` is set to True, after a simulator run a file named `blocks.db` will be written to the `osmosis/tests/simulator/` folder. This database file provides the following information:

Screen Shot 2022-08-15 at 7 19 13 PM

## Brief Changelog

- Adds sqlite to simulator
- Adds flag `WriteStatsToDB`
- Adds basic gas info to new messages (legacy messages had been added through an SDK PR)
- Adds the following data struct used by SQL:
  - blocks(height, module, name, comment, passed, gasWanted, gasUsed)

## Testing and Verifying

This change is already covered by existing tests

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? no
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? no
  - How is the feature or change documented? not applicable

### CODE REVIEW

File: tests/simulator/sim_test.go [hunk 1]
```diff
@@ -188,6 +212,7 @@ func TestAppStateDeterminism(t *testing.T) {
 				initFns,
 				osmosis.SimulationManager().Actions(config.Seed, osmosis.AppCodec()), // Run all registered operations
 				config,
+				sqlDB,
```
Comment (p0mvn) [inline tests/simulator/sim_test.go:215]: nit: consider providing nil as the arg since the variable is never initialized

**Review by p0mvn — APPROVED** [review]

File: tests/simulator/sim_test.go [hunk 2]
```diff
@@ -45,14 +47,34 @@ func TestFullAppSimulation(t *testing.T) {
...
 	}
+	if config.WriteStatsToDB {
+		sqlDB, err = sql.Open("sqlite3", "./blocks.db")
+		if err != nil {
+			tb.Fatal(err)
+		}
+		defer sqlDB.Close()
+		sts := `
+	DROP TABLE IF EXISTS blocks;
+	CREATE TABLE blocks (id INTEGER PRIMARY KEY, height INT,module TEXT, name TEXT, comment TEXT, passed BOOL, gasWanted INT, gasUsed INT);
```
Comment (ValarDragon) [inline tests/simulator/sim_test.go:68]: The table structure should be defined / encapsulated in the simulator

**Review by ValarDragon — APPROVED** [review]

File: tests/simulator/sim_test.go [hunk 3]
```diff
@@ -102,6 +114,7 @@ func fullAppSimulation(tb testing.TB, is_testing bool) {
 		initFns,
 		osmosis.SimulationManager().Actions(config.Seed, osmosis.AppCodec()), // Run all registered operations
 		config,
+		sqlDB,
```
Comment (ValarDragon) [inline tests/simulator/sim_test.go:117]: Wait why are we passing in a new db in the simulator instantiation? The simulator itself should handle creating and cleaning up the entire db

**Comment by ValarDragon** [comment]
Err, can you add blocks.db to the gitignore and delete it from repo! Happy to merge this after that, and we can do followups subsequently

File: tests/simulator/sim_test.go (hunk 3 above)
Comment (czarcas7ic) [reply tests/simulator/sim_test.go:117]: Sorry I misunderstood your original comment! I had decided to do it at the highest level to ensure I don't close the DB prematurely but on further review it should be a non issue if I just defer this within the simulateFromSeed function itself

**Review by ValarDragon — APPROVED** [review]

**Pull request merged**